Navigation/AdminMenu.cs

```diff
@@ -0,0 +1,31 @@
+using Lombiq.Privacy.Permissions;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Navigation;
+using System;
+using System.Threading.Tasks;
+using static Lombiq.Privacy.Constants.GroupIds;
+
+namespace Lombiq.Privacy.Navigation
+{
+    public class AdminMenu : INavigationProvider
+    {
+        private readonly IStringLocalizer T;
+
+        public AdminMenu(IStringLocalizer<AdminMenu> localizer) => T = localizer;
+
+        public Task BuildNavigationAsync(string name, NavigationBuilder builder)
+        {
+            if (!string.Equals(name, "admin", StringComparison.OrdinalIgnoreCase)) return Task.CompletedTask;
```

Comment:
`EqualsOrdinalIgnoreCase()` kellene itt is.

Reply:
Egy gyors kérdés a `EqualsOrdinalIgnoreCase()`-el kapcsolatban.
Jelelnleg a következőképpen müködik
```csharp
public static bool EqualsOrdinalIgnoreCase(this string text, string? value) =>
            text.Equals(value, StringComparison.OrdinalIgnoreCase);
```
Nem lenne jobb az alábbi? Mert az első esetben ha a `string ` `null` akkor elszáll a program `NullReferenceException`-nel (legalábbis ha jól emlékszem ezzel)
```csharp
public static bool EqualsOrdinalIgnoreCase(this string text, string? value) =>
            text.Equals(value, StringComparison.OrdinalIgnoreCase);
```

Reply:
Másodiknál nem? Mert fura lenne, ha ez a kettő másképp működne.

Az elgondolás ezeknél a shotcutoknál az, hogy csak pontosan azt csinálják, ami a nevükben van. Szóval ha az eredeti method NRE-t ad, akkor itt is az lesz.

Reply:
Most újra próbáltam az alábbi kódot és valóbban van külömbség. A `string.Equals()` nullable string paraméterekkel dolgozik mig a változó ha null akkor érthető módon nem lehet rajta elvégezni az Equals() müveletet.

```csharp
 string test = null;

 var resultOnStringCase = string.Equals(test, "Test string", StringComparison.OrdinalIgnoreCase); //resultOnStringCase: false
 var resultOnVariableCase = test.Equals("Test string", StringComparison.OrdinalIgnoreCase); //resultOnVariableCase: NullReferenceException
```
 Ez akkor tünt fel amikor előszőr javasoltad a HelpfulLibet és kicsieréltem azáltalam irt `string.Equals()`-t `EqualsOrdinalIgnoreCase()`-re és emiatt be kellet raknom egy plusz null checket mert elszált a program.

Reply:
Aha, ez fura. Mindenesetre mivel itt mindegyik method instance-levelt proxyzik, nem staticot, maradnia kell.